### PR TITLE
[MU4] Dock system stabilization

### PR DESCRIPTION
**Pull request by Eism**
- Disabled restoring app geometry when switching pages
- Corrected sizes of panels

### CODE REVIEW

File: src/appshell/view/dockwindow_kdab/dockwindow.cpp [hunk 1]
```diff
@@ -181,15 +184,24 @@ DockPage* DockWindow::pageByUri(const QString& uri) const
 
 void DockWindow::saveState(const QString& pageName)
 {
-    configuration()->setPageState(pageName.toStdString(), m_layoutSaver->serializeLayout());
+    KDDockWidgets::LayoutSaver layoutSaver;
+    configuration()->setPageState(pageName.toStdString(), layoutSaver.serializeLayout());
```
Comment (igorkorsukov) [inline src/appshell/view/dockwindow_kdab/dockwindow.cpp:188]: How is data passing to LayoutSaver that needs to be serialized?

File: src/appshell/view/dockwindow_kdab/dockwindow.cpp [hunk 2]
```diff
@@ -181,15 +184,24 @@ DockPage* DockWindow::pageByUri(const QString& uri) const
...
 }
 
-void DockWindow::restoreState(const QString& pageName)
+void DockWindow::restoreState(const QString& pageName, bool restoreGeometry)
 {
     QByteArray state = configuration()->pageState(pageName.toStdString());
-    if (!state.isEmpty()) {
-        m_layoutSaver->restoreLayout(state);
+    if (state.isEmpty()) {
```
Comment (igorkorsukov) [inline src/appshell/view/dockwindow_kdab/dockwindow.cpp:194]: Probably should be `if (!state.isEmpty())`

File: src/appshell/view/dockwindow_kdab/dockwindow.cpp [hunk 3]
```diff
@@ -181,15 +184,24 @@ DockPage* DockWindow::pageByUri(const QString& uri) const
...
+        return;
+    }
+
+    KDDockWidgets::RestoreOption option = KDDockWidgets::RestoreOption::RestoreOption_RelativeToMainWindow;
+    if (restoreGeometry) {
+        option = KDDockWidgets::RestoreOption::RestoreOption_None;
     }
+
+    KDDockWidgets::LayoutSaver layoutSaver(option);
+    layoutSaver.restoreLayout(state);
```
[the remark this answers is not in the document]
Comment (Eism) [reply src/appshell/view/dockwindow_kdab/dockwindow.cpp:204]: Yes, LayoutSaver uses DockRegistry singelton

[the remark this answers is not in the document]
Comment (igorkorsukov) [reply src/appshell/view/dockwindow_kdab/dockwindow.cpp:204]: Ok, I got it.

**Review by igorkorsukov — APPROVED** [review]

**Pull request merged**